### PR TITLE
[38693] Scroll to oldest unread activity in the activity tab

### PR DESCRIPTION
**Pull request by bsatarnejad**
- make the tabs stick to the top while scrolling.
- scroll to the oldest unread notification by considering deep link including an activity anchor, and leave some space for the activity before or after it.

https://community.openproject.org/projects/openproject/work_packages/38693/activity

### CODE REVIEW

File: frontend/src/app/shared/components/tabs/scrollable-tabs/scrollable-tabs.component.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<div class="op-scrollable-tabs" [ngClass]="classes">
+<div class="op-scrollable-tabs -sticky" [ngClass]="classes">
```
Comment (b12f) [inline frontend/src/app/shared/components/tabs/scrollable-tabs/scrollable-tabs.component.html]: It's good if the parent sets the positioning of an element, that way a BEM block is independent of the place it is being used in. So I think it's better if you add this modifier to the parent context. Also, modifiers should be done with `&_sticky`, not `.-sticky`.

File: frontend/src/app/features/work-packages/components/wp-single-view-tabs/activity-panel/activity-base.controller.ts [hunk 2]
```diff
@@ -141,6 +142,23 @@ export class ActivityPanelBaseController extends UntilDestroyedMixin implements
     return !!this.notifications.find((notification) => notification._links.activity?.href === activityHref);
   }
 
+  protected scrollToUnreadNotification():void {
+    // scroll to the unread notification only if there is no deep link
+    if (!(window.location.href.indexOf('activity#') > -1)) {
```
Comment (b12f) [inline frontend/src/app/features/work-packages/components/wp-single-view-tabs/activity-panel/activity-base.controller.ts]: To reduce nesting, I'd recommend using the return-early pattern:

```
if (window.location.href.indexOf('activity#') > -1) {
  return
}
```

Same for the `if` statements below

File: frontend/src/app/features/work-packages/components/wp-single-view-tabs/activity-panel/activity-base.controller.ts [hunk 3]
```diff
@@ -141,6 +142,23 @@ export class ActivityPanelBaseController extends UntilDestroyedMixin implements
...
+      const unreadNotifications = document.querySelectorAll('[data-qa-selector="user-activity-bubble"]');
```
Comment (b12f) [inline frontend/src/app/features/work-packages/components/wp-single-view-tabs/activity-panel/activity-base.controller.ts]: I'm not sure if we should use `qa` selectors for anything other than testing. @opf/frontend what do you think?

Comment (bsatarnejad) [reply frontend/src/app/features/work-packages/components/wp-single-view-tabs/activity-panel/activity-base.controller.ts]: Hi @b12f  thanks for your review, @HDinger also said that we use data-qa-selector for testing issues. I will use class names then.

**Review by b12f — APPROVED** [review]

**Pull request merged**